extension/url-handler.js

```diff
@@ -0,0 +1,98 @@
+
+var urlHandler = {};  // Namespace
+
+/**
+ * In most cases, we don't want to display a full URL with a path.
+ * @param {inputUrl} The original url.
+ * @returns A more minimized version of the url.
+ */
+urlHandler.GetMinimalUrl = function(inputUrl) {
+  var scheme = '';
+  var hostport = '';
+  var parser = document.createElement('a');
+  parser.href = inputUrl;
+
+  if (parser.protocol === 'chrome-extension:') {
+    // The parser interprets some urls without schemes as relative to
+    // chrome-extension:. If that happens, turn it into a https url instead.
+    parser.href = 'https://' + inputUrl;
+  }
+  if ((parser.protocol === 'file:' || parser.protocol === ':') && 
+      (inputUrl.substring(0, 6).indexOf(':') == -1)) {
```

Comment:
Good question. It works with chrome: because the bounds are (0, 6) and not (0, 5). To prove it to myself I added a test case for chrome://interstitials.